src/dali_executor/utils/utils.h

```diff
@@ -102,6 +102,27 @@ std::vector<TensorListShape<Dims>> split_list_shape(const TensorListShape<Dims>
   return result;
 }
 
+
+inline TensorListShape<> split_outer_dim(const TensorListShape<> &shape) {
+  auto in_ndim = shape.sample_dim();
+  ENFORCE(in_ndim >= 2, "To split outer dimension, batch must be at leat 2-dimensional.");
```

Comment:
It may not be particularly real life use case, but why not allow scalars after the split?